driver-sync/src/test/functional/com/mongodb/client/AbstractSessionsProseTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client;
+
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.UpdateOneModel;
+import com.mongodb.client.model.Updates;
+import com.mongodb.event.CommandListener;
+import com.mongodb.event.CommandStartedEvent;
+import org.bson.BsonDocument;
+import org.bson.Document;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import static com.mongodb.ClusterFixture.getConnectionString;
+import static com.mongodb.ClusterFixture.getDefaultDatabaseName;
+import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+// Prose tests from https://github.com/mongodb/specifications/tree/master/source/sessions
+public abstract class AbstractSessionsProseTest {
+
+    protected abstract MongoClient getMongoClient(MongoClientSettings settings);
+
+    // Test 13
+    @Test
+    public void shouldCreateServerSessionOnlyAfterConnectionCheckout() throws InterruptedException {
+        assumeTrue(serverVersionAtLeast(3, 6));
+        // given
+        Map<BsonDocument, Boolean> lsidSet = new ConcurrentHashMap<>();
```

Comment:
[nit]

We can do `Set<BsonDocument> lsidSet = ConcurrentHashMap.newKeySet();`, and get a concurrent `Set` instead of a `Map`.

Reply:
Or we can just remove this test entirely... :)

Reply:
💪